cmd/drone-docker/main.go

```diff
@@ -242,6 +244,7 @@ func run(c *cli.Context) error {
 		},
 		Build: docker.Build{
 			Remote:      c.String("remote.url"),
+			Ref:         c.String("commit.sha"),
```

Comment:
the ref should be set to `c.String("commit.ref")`

Reply:
The goal here was to not pollute the `org.label-schema.vcs-ref` label with the random ID, so I copied "Name" which was the sha hash. If the term is confusing, maybe it should be called `CommitSha`?

Reply:
the ref is not a random string, it is a git ref, for example `refs/heads/master`

Reply:
That's true, however it would break existing behavior as this label contained the commit sha before the change by using the `Name` value.

Reply:
I see, `org.label-schema.vcs-ref` is supposed to be the commit sha.  I recommend removing this and sending a separate pull request. This requires additional discussion. We can merge the random Name in the short term.

Reply:
If I remove it, it would be placing a random ID that is not a sha hash into the label, which I think would be as debatable as emptying it. Of course it only applies to events that don't supply the sha and are crashing today, like gogs.

Looking up the schema it states http://label-schema.org/rc1/
> All labels are OPTIONAL, however if present MUST be prefixed with the namespace org.label-schema.
> ...
> vcs-ref | org.label-schema.vcs-ref = "279FA63" | Identifier for the version of the source code from which this  image was built. For example if the version control system is git this  is the SHA.

Perhaps the best option is not to label with an empty string, but not label at all. As this would comply with the spec and match how gogs lacks the sha in the event.

Reply:
Just add an SHA attribute and rename Ref to it. That's not so much confusing. Beside this change from Name to Ref/SHA you should not change the label schema, we still got to discuss how to continue with that since label-schema.org is deprecated.

Reply:
I'll hold of for this discussion to happen then. As it will inevitably change when the SHA isn't available.

Reply:
I think the proper thing to do would be to get Gogs to send the SHA. As a sidenote Gitea should send the SHA (if not let me know and I'll send a PR to fix that).

Reply:
@techknowlogick that would be awesome! Last I checked Gitea did not send this information in the payload. See https://github.com/drone/go-scm/blob/master/scm/driver/gitea/testdata/webhooks/tag_create.json

Reply:
Just tested and there is a key in the webhook response labled: `sha`. Let me send a PR to go-scm to add this information.

Reply:
PR is here: https://github.com/drone/go-scm/pull/22

Reply:
Thank you for a great fix there @techknowlogick.
I had a dig to see if the same would apply to gogs, but unfortunately it seems there's no sha here.
https://github.com/gogs/go-gogs-client/blob/5898069c37e1044229e3853adba654d61e45816b/repo_hook.go#L114

<details><summary>Example tag payload</summary>

```json
{
    "ref": "v2.4.3",
    "ref_type": "tag",
    "default_branch": "master",
    "repository": {
        "id": 1,
        "owner": {
            "id": 1,
            "username": "example",
            "login": "example",
            "full_name": "",
            "email": "gogs@example.com",
            "avatar_url": "https://git.example.com/avatars/1"
        },
        "name": "hello-cicd",
        "full_name": "example/hello-cicd",
        "description": "",
        "private": true,
        "fork": false,
        "parent": null,
        "empty": false,
        "mirror": false,
        "size": 1300480,
        "html_url": "https://git.example.com/example/hello-cicd",
        "ssh_url": "ssh://git@git.example.com/example/hello-cicd.git",
        "clone_url": "https://git.example.com/example/hello-cicd.git",
        "website": "",
        "stars_count": 0,
        "forks_count": 0,
        "watchers_count": 1,
        "open_issues_count": 0,
        "default_branch": "master",
        "created_at": "2018-04-17T20:45:00Z",
        "updated_at": "2019-04-19T09:20:56Z"
    },
    "sender": {
        "id": 1,
        "username": "example",
        "login": "example",
        "full_name": "",
        "email": "gogs@example.com",
        "avatar_url": "https://git.example.com/avatars/1"
    }
}
```

</details>

Reply:
I've opened the PR there, we'll have to see what they feel about this change.
https://github.com/gogs/gogs/pull/5689